gen3utils/etl/etl_validator.py

```diff
@@ -1,202 +1,275 @@
+            )
+            if edge not in list_of_nodes:
+                recorded_errors.append(PathError(path))
+            if str_fields is not None:
+                fields = str_fields.split(",")
+                for f in fields:
+                    src = f.split(":")[-1] if f.find(":") != -1 else f
+                    name = f.split(":")[0] if f.find(":") != -1 else f
+                    names.append(
+                        validate_name_src(
+                            {"name": name, "src": src},
+                            edge,
+                            recorded_errors,
+                            nodes_with_props,
+                        )
+                    )
+    return names
+
+
+def get_all_nodes(model):
+    labels_to_back_refs = {}
+    nodes_with_props = {}
+    categories_to_labels = {}
+    all_classes = model.Node.get_subclasses()
+    for n in all_classes:
+        present_node = n._pg_edges
+        present_props = n.__pg_properties__
+        present_node_props = [k for k in present_props.keys()]
+        present_node_props.append("id")
+        category = n._dictionary.get("category")
+        if category not in categories_to_labels:
+            categories_to_labels[category] = []
+        categories_to_labels.get(category).append(n.label)
+        backref = ""
+        if len(present_node) > 0:
+            backref = list(present_node.values())[0].get("backref")
+            labels_to_back_refs[n.label] = backref
+        nodes_with_props.update({backref: present_node_props})
+    return labels_to_back_refs, nodes_with_props, categories_to_labels
+
+
+def validate_mapping(dictionary_url, mapping_file):
+    dictionary, model = init_dictionary(dictionary_url)
+    with open(mapping_file) as f:
+        mappings = yaml.load(f, Loader=yaml.SafeLoader)
+
+    labels_to_back_refs, nodes_with_props, categories_to_labels = get_all_nodes(model)
+
+    recorded_errors = []
+    indices = {}
+    for m in mappings.get("mappings"):
+        index = Index(m.get("doc_type"))
+        indices[index.name] = index
+        category = m.get("category")
+        if category is not None:
+            first_categorized_node = categories_to_labels.get(category)[0]
+        else:
+            first_categorized_node = m.get("doc_type")
+        for key, value in m.items():
+            if key.find("props") != -1:
+                root_path = (
+                    labels_to_back_refs.get(first_categorized_node)
+                    if key == "props"
+                    else None
+                )
+                validate_list_props(
+                    value,
+                    labels_to_back_refs,
+                    nodes_with_props,
+                    recorded_errors,
+                    root_path,
+                    index,
+                )
+    for m in mappings.get("mappings"):
+        for k, v in m.items():
+            validate_joining_list_props(v, recorded_errors, indices)
```

Comment:
why not do this instead, and simplify `validate_joining_list_props` to not iterate through all the keys?
```
for m in mappings.get("mappings"):
    joining_props = m.get("joining_props", [])
    validate_joining_list_props(joining_props, recorded_errors, indices)
```